src/main/java/org/beanpod/switchboard/util/MaintainDeviceStatus.java

```diff
@@ -0,0 +1,93 @@
+package org.beanpod.switchboard.util;
+
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+import lombok.RequiredArgsConstructor;
+import org.beanpod.switchboard.dao.DeviceDaoImpl;
+import org.beanpod.switchboard.dto.DecoderDto;
+import org.beanpod.switchboard.dto.EncoderDto;
+import org.beanpod.switchboard.dto.StreamDto;
+import org.beanpod.switchboard.dto.mapper.DeviceMapper;
+import org.beanpod.switchboard.entity.DummyInterface;
+import org.beanpod.switchboard.service.LogService;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class MaintainDeviceStatus {
```

Comment:
After looking more closely at this class and considering our requirements, there's a lot of logic I feel that could be reduced/avoided if we follow how @mohhef tackled logging using AOP. @sebhani, what is your gut telling you?

Reply:
I've extracted the logging functionality to an Aspect class.